Loop/Managers/ServicesManager.swift

```diff
@@ -7,53 +7,114 @@
 //
 
 import LoopKit
+import LoopKitUI
 
 protocol ServicesManagerObserver {
 
-    /// The service manager update the list of available services.
+    /// The service manager updated the list of active services.
     ///
-    /// - Parameter services: The list of available services
-    func servicesManagerDidUpdate(services: [Service])
+    /// - Parameter activeServices: The list of active services.
+    func servicesManagerDidUpdate(activeServices: [Service])
 
 }
 
 class ServicesManager {
 
     private let queue = DispatchQueue(label: "com.loopkit.ServicesManagerQueue", qos: .utility)
 
-    private let lock = UnfairLock()
+    private let pluginManager: PluginManager
+
+    private var services = [Service]()
+
+    private let servicesLock = UnfairLock()
 
     private var observers = WeakSet<ServicesManagerObserver>()
 
-    var services: [Service] {
-        didSet {
-            dispatchPrecondition(condition: .onQueue(.main))
-            UserDefaults.appGroup?.services = services
-            notifyObservers()
+    private let observersLock = UnfairLock()
```

Comment:
Interesting that the observers are also locked. Should this be common practice? I have noticed other places (e.g., DexcomCGMKit) where this isn't the case.

Reply:
I'd think so.